### ISSUE TITLE
Icewm does not start from any welcome screen [bug]

### ISSUE THREAD

**Issue by ghost** [bbidulock/icewm#155]
My system:
```
Linux arch 4.12.4-1-ARCH #1 SMP PREEMPT Fri Jul 28 18:54:18 UTC 2017 x86_64 GNU/Linux
```
When you try to sign in to icewm, you will be kicked back to the welcome screen.
I found errors only in the file .xsession-errors:
```
icewm-session: using /home/user/.icewm for private configuration files
icewmtray: symbol lookup error: icewmtray: undefined symbol: xlib_rgb_init
icewmbg: symbol lookup error: icewmbg: undefined symbol: xlib_rgb_init
icewm: symbol lookup error: icewm: undefined symbol: xlib_rgb_init
icewmbg: symbol lookup error: icewmbg: undefined symbol: xlib_rgb_init
```
Help me understand and complete the instructions if I do not have some library

**Comment by gijsbers** [comment]
Which kind of Linux distribution are you using?
It looks as if your  libgdk-pixbuf2.0 is not installed.
Try something like:

    apt-get  libgdk-pixbuf2.0

**Comment by mufranoid** [comment]
No idea what broke (or was intentionally scrapped/changed) in gdk-pixbuf2 2.36.8, but downgrading to 2.36.7 should be a temporary solution.

**Comment by bbidulock** [comment]
From what I can tell, 2.36.8 PKGBUILD uses meson build instead of autoconf/automake.  It appears that the meson build is not building libgdk_pixbuf_xlib-2.0.so correctly and it is missing the necessary symbols.  There is a flyspray report at https://bugs.archlinux.org/task/55097

It is not an icewm problem.

**Comment by bbidulock** [comment]
Fixed.  I built a gdk-pixbuf2-xlib package and pushed it to the AUR that uses autoconf build for 2.36.8 instead of meson and the symbols are restored.  Also pushed an icewm2 and icewm-git version that depends on gdk-pixbuf2-xlib instead of gdk-pixbuf2.  Rebuild and everything should work properly again.

**Comment by ghost** [comment]
> Which kind of Linux distribution are you using?
> It looks as if your libgdk-pixbuf2.0 is not installed.
> Try something like:

In a post it is written that at me for system. Pacman can not find this package:
```
Error: target not found: libgdk-pixbuf2.0
```

**Comment by ghost** [comment]
> Fixed. I built a gdk-pixbuf2-xlib package and pushed it to the AUR that uses autoconf build for 2.36.8 instead of meson and the symbols are restored. Also pushed an icewm2 and icewm-git version that depends on gdk-pixbuf2-xlib instead of gdk-pixbuf2. Rebuild and everything should work properly again.

I have a weak computer and I will compile the source for a very long time. I have already started but it's probably for 12 hours. Do you have any binaries?

**Comment by bbidulock** [comment]
You can add the following to pacman.conf:

[custom]
Server = http://www.unexicon.com/repo/pkgs/arch/os/$arch/$repo

All my AUR packages are there.  Or you can just go after the binaries you want at:

http://www.unexicon.com/repo/pkgs/arch/os/x86_64/custom/gdk-pixbuf-0.22.0-x86_64.pkg.tar.xz
http://www.unexicon.com/repo/pkgs/arch/os/x86_64/custom/icewm2-1.4.2-4-x86_64.pkg.tar.xz
http://www.unexicon.com/repo/pkgs/arch/os/x86_64/custom/icewm-git-1.4.2.87-1-x86_64.pkg.tar.xz

But my upload speed here is slow (1M).

**Comment by bbidulock** [comment]
@gijsbers thinks  your running something like debian and thus the apt command.  I know your running Arch  because I use it exclusively myself (for about 5 years now).  I like developing against up to date software rather than something that is 18 months old already.

**Comment by Code7R** [comment]
@bbidulock C'mon, seriously, you should know better than that. 

First, the "apt-get" command above was wrong (it should be apt-get install ...) so I don't think he is running a Debian derivate. However, since the reports talk about "compiling" I guess they are using Gentoo or LFS or Yocto or some other source-based distro.

Second, FYI: most popular "something like Debian" is called Ubuntu which has 6-month release cycle for non-LTS branch. Even Debian has a branch called Testing which is only a few days/weeks behind bleeding edge. Despite the name, it's what comes close in terms of quality to what some other distros call "stable" (do the math).

Third: we have explicit policies on symbol versioning. Crude ABI breaks are rare, especially in Testing, especially for popular libs like pixbuf. The most likely reason to get in a situation where deps are missing is by breaking the installation somehow (OOM or OO disk space), in which case the better command to resolve the apt problem would be just "apt-get install" - that should re-resolve missing dependencies and fix the setup.

**Comment by ghost** [comment]
> You can add the following to pacman.conf:
> [custom]
> Server = http://www.unexicon.com/repo/pkgs/arch/os/$arch/$repo
It all worked, the problem closes

**Comment by bbidulock** [comment]
@morteryler, Arch has released gdk-pixbuf2-2.36.8-2 that fixes the meson/ninja build and now works again.  I will flip the dependency back on AUR.

@Code7R, sorry I didn't mean to open a distro debate.  Debian would be my second choice anyhoo.
